src/core/server/plugins/types.ts

```diff
@@ -195,6 +199,19 @@ export interface Plugin<
   stop?(): void;
 }
 
+export const SharedGlobalConfigKeys = {
+  // We can add more if really needed
+  kibana: ['defaultAppId', 'index'] as const,
+  elasticsearch: ['shardTimeout', 'requestTimeout', 'pingTimeout', 'startupTimeout'] as const,
+  path: ['data'] as const,
+};
+
+export type SharedGlobalConfig = RecursiveReadonly<{
```

Comment:
not sure if we should call `deepFreeze` on the config to make really `RecursiveReadonly`